src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -78,10 +78,10 @@ public function __construct(string $name, $shortcut = null, int $mode = null, st
                 $shortcut = implode('|', $shortcut);
             }
             $shortcuts = preg_split('{(\|)-?}', ltrim($shortcut, '-'));
-            $shortcuts = array_filter($shortcuts);
+            $shortcuts = array_filter($shortcuts, 'strlen');
             $shortcut = implode('|', $shortcuts);
 
-            if (empty($shortcut)) {
+            if ($shortcut === '') {
```

Comment:
This will change the code's behavior and throw the `InvalidArgumentException` when `$shortcut` is nonempty. 

Perhaps you meant to suggest:

```php
if ('' === $shortcut) {
```

To "flip" the expression into a "yoda condition"?

Reply:
Oops, sorry, my bad, I meant: use Yoda condition

Reply:
I have changed it in bf8cbbd.